internal/activity.go

```diff
@@ -264,6 +264,12 @@ func RecordActivityHeartbeat(ctx context.Context, details ...interface{}) {
 	getActivityOutboundInterceptor(ctx).RecordHeartbeat(ctx, details...)
 }
 
+// GetClient returns a client that can be used to interact with the Temporal
+// service from an activity.
+func GetClient(ctx context.Context) Client {
+	return getActivityEnv(ctx).client
```

Comment:
@cretz should this be intercept-able?

Reply:
My allow intercepting things like getting the loggers or metrics so it seems like we would allow this, but if we didn't in other SDKs I am fine to leave off.